packages/devtools_app/lib/src/network/network_screen.dart

```diff
@@ -158,8 +158,8 @@ class _NetworkScreenBodyState extends State<NetworkScreenBody>
 
   @override
   void dispose() {
-    _networkController?.stopRecording();
     super.dispose();
+    _networkController?.stopRecording();
```

Comment:
calls to super dispose should always be first. Lets try to fix this a different way.

Reply:
Cancelled listeners in `AutoDisposeMixin` in `deactivate` instead of `dispose`. This ensures the listeners are cancelled before we do anything in `dispose` that may trigger listeners we added in `didChangeDependencies` via `AutoDisposeMixin.addAutoDisposeListener`